### DISCUSSION TITLE
Lack of blast matches

### DISCUSSION THREAD

**Discussion by meganewing** [sr320/course-fish546-2021#129] [Q&A] [answered]
I have 12 genes that are significantly different for my samples, but when I ran blast only 1 of them came up with a match. I even tried looking some up on the blast website using smart blast in case it was something I did wrong with my blast code, but to no avail. 
What does this mean?

**Comment by meganewing** [comment]
The genes are:

ENSOKIT00005039714.1
ENSOKIT00005060882.1
ENSOKIT00005062889.1
ENSOKIT00005063078.1
ENSOKIT00005065079.1
ENSOKIT00005068341.1
ENSOKIT00005078997.1
ENSOKIT00005091707.1
ENSOKIT00005113439.1
ENSOKIT00005117433.1
ENSOKIT00005125389.1

if that helps

**Comment by kubu4** [comment]
Please share code you ran.

**Reply by meganewing** [reply]
blast: https://github.com/fish546-2021/Megan-project/blob/main/code/step8-blast.ipynb

**Comment by sr320** [comment]
can you provide a fasta file that contains the 12 genes?

**Reply by meganewing** [reply]
My files are fastq not fasta if that works? Here's both read ends of one of the control treatment samples. 
The gene names were what were given to me after doing Kallisto. 

[control read1](https://gannet.fish.washington.edu/fish546_2021/Files_Giles_Uploaded_For_Chris/17104-02RT-01-7_S15_L002_R1_001.fastq.gz)

[control read2](https://gannet.fish.washington.edu/fish546_2021/Files_Giles_Uploaded_For_Chris/17104-02RT-01-7_S15_L002_R2_001.fastq.gz)

**Reply by sr320** [reply]
your reference is fasta... the 12 genes are in there.

**Reply by meganewing** [reply]
here's my reference file. Was it something I did wrong with my blast? Because when I ran it, all 12 genes weren't in there.
[Oncorhynchus_kisutch.Okis_V2.ncrna.fa.gz](https://github.com/sr320/course-fish546-2021/files/6117544/Oncorhynchus_kisutch.Okis_V2.ncrna.fa.gz)

**Reply by sr320** [reply]
I am asking you to provide a fasta of just the 12 genes, not the entire transcriptome.

**Reply by meganewing** [reply]
I don't have a fasta of just the 12 genes

**Reply by kubu4** [reply]
I've extracted the 11 sequences you provided in your original post.

I've add the FastA to your repo's data folder. Here's the link:

https://github.com/fish546-2021/Megan-project/blob/main/data/Oncorhynchus_kisutch.Okis_V2.ncrna_11-DEGs.fa

I accomplished by using the `seqkit` software, but I think you can also do it with `samtools` (this is an example of the utility of having a FastA index file; that's what the `faidx` part of the command works on):

```bash
seqkit-0.15.0 faidx \
Oncorhynchus_kisutch.Okis_V2.ncrna.fa \
ENSOKIT00005039714.1 \
ENSOKIT00005060882.1 \
ENSOKIT00005062889.1 \
ENSOKIT00005063078.1 \
ENSOKIT00005065079.1 \
ENSOKIT00005068341.1 \
ENSOKIT00005078997.1 \
ENSOKIT00005091707.1 \
ENSOKIT00005113439.1 \
ENSOKIT00005117433.1 \
ENSOKIT00005125389.1 \
> Oncorhynchus_kisutch.Okis_V2.ncrna_12-DEGs.fa
```

Go ahead and BLAST that FastA and let us know how it goes.

I'll try BLASTing in a bit and report back, as well.

**Reply by kubu4** [reply]
Alrighty, I ran BLASTx and got the same results as you - no matches (see bottom near bottom line of screencap):

![image](https://user-images.githubusercontent.com/4514104/110705256-ff257380-81aa-11eb-9c47-5d297b80c7ba.png)


If you relax the e-value (e.g. 1E-4), you'll get a _single_ sequence from your query which has some matches.

That's all I have for now.

Since I'm not totally clear on how your FastA file (or, where it came from) ties into your gene expression analysis, it's hard for me to explain what might be going on.

**Reply by meganewing** [reply]
I got pretty much the same except got two sequences that had matches. 
Do you think it could it just be that these other genes aren't described yet?

**Comment by sr320** [comment]
I can assist if you can make one.

**Reply by meganewing** [reply]
How do I make one? Right now the names are just listed in a .tab file

**Reply by sr320** [reply]
Easiest is open reference and copy / paste the 12 sea to new file.

**Reply by meganewing** [reply]
I don't know the sequences, just the names. Could we arrange a zoom to discuss?

**Comment by sr320** [comment, marked as answer]
I did a blastn on these sequences on the web.

and for a majority of which we can assign a putative ID
see https://d.pr/v/NEJ3B8

**Reply by meganewing** [reply]
Oh marvelous. That's what I was trying to do last night but was using blastp. Wasn't quite sure what the difference was, but this is extremely helpful. Thank you!!!

**Reply by kubu4** [reply]
BLASTp = Protein query vs. protein database

BLASTn = Nucleotide query vs. nucleotide database

BLASTx = Nucleotide query vs. protein database (query is translated into six different reading frames)

tBLASTn = Protein query vs. nucleotide database (database is translated into six different reading frames)

tBLASTx = Nucleotide query vs. nucleotide database (both query and database are translated in to six different reading frames and compared)
